Dokan admin dashboard vendor profile re-design ( Base PR )

Aunshon opened the pull request:
### All Submissions:

* [ ] My code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)
* [ ] My code satisfies feature requirements
* [ ] My code is tested
* [ ] My code passes the PHPCS tests
* [ ] My code has proper inline documentation
* [ ] I've included related pull request(s) (optional)
* [ ] I've included developer documentation (optional)
* [ ] I've added proper labels to this pull request

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

### Related Pull Request(s)
* https://github.com/getdokan/dokan/pull/2926
* https://github.com/getdokan/dokan-pro/pull/4604
* https://github.com/getdokan/dokan-pro/pull/4884

### Merged Pull Request(s)
* https://github.com/getdokan/dokan/pull/2890
* https://github.com/getdokan/dokan/pull/2819
* https://github.com/getdokan/dokan/pull/2889
* https://github.com/getdokan/dokan/pull/2766


### Closes 
* Closes https://github.com/getdokan/dokan-pro/issues/4424
* Closes https://github.com/getdokan/plugin-internal-tasks/issues/594
* Closes https://github.com/getdokan/dokan-pro/issues/4475


### How to test the changes in this Pull Request:

* Steps or issue link


### Changelog entry

*Title*

Detailed Description of the pull request. What was previous behaviour
and what will be changed in this PR.


### Before Changes

Describe the issue before changes with screenshots(s).


### After Changes

Describe the issue after changes with screenshot(s).


### Feature Video (optional)

Link of detailed video if this PR is for a feature.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Full admin vendor management UI: create, edit, and vendor details pages with routing.
  * Vendor payment management and withdrawal flows with per-method modals and save actions.
  * Vendor store data and analytics (top-selling products) surfaced in the admin.

* **UI Improvements**
  * New payment method icons and richer payment UI.
  * Loading skeletons and improved media/image handling for vendor forms and banners.
  * Enhanced vendor list actions and navigation (create/edit flows).
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

shohag121 committed:
feat: add VendorsSingle page and routing to admin dashboard

shohag121 committed:
Merge remote-tracking branch 'refs/remotes/upstream/develop' into update/admin-vendor-profile

shohag121 committed:
WIP:  Vendor Store

shohag121 committed:
WIP:  Vendor single display header section

shohag121 committed:
WIP: Vendor single display Sidebar section

shohag121 committed:
Merge branch 'develop' into update/admin-vendor-profile

# Conflicts:
#	assets/js/vue-admin.js
#	assets/js/vue-bootstrap.js

shohag121 committed:
enhance: Add social links and email functionality to vendor profile

MdAsifHossainNadim committed:
enhance: Implement send email button and popup for vendor communication.

MdAsifHossainNadim committed:
enhance: Redesign vendor overview with tabbed interface and new card components

MdAsifHossainNadim committed:
enhance: Refactor vendor tabs to use hooks for dynamic configuration

MdAsifHossainNadim committed:
enhance: Integrate top products fetching and loading state in vendor overview

Aunshon committed:
enhance: Remove unused tabs from vendor overview and streamline tab configuration.

Aunshon committed:
Merge remote-tracking branch 'origin/enhance/vendor-overview-redesign-with-tabs-ui' into enhance/vendor-overview-redesign-with-tabs-ui

MdAsifHossainNadim committed:
enhance: Remove unused Badges tab and optimize vendor stats fetching in tab sections

MdAsifHossainNadim committed:
enhance: Add position property to tabs and implement sorting for dynamic tab order

MdAsifHossainNadim committed:
enhance: Update ProductCard and StatCard to use dangerouslySetInnerHTML for dynamic content rendering

MdAsifHossainNadim committed:
enhance: Refactor InfoCard to use slots for email functionality integration

MdAsifHossainNadim committed:
enhance: Refactor InfoCard to use slots for email functionality integration

MdAsifHossainNadim committed:
Merge branch 'enhance/implement-send-email-button-and-popup-for-vendor-redesign' into enhance/vendor-overview-redesign-with-tabs-ui

# Conflicts:
#	src/admin/dashboard/pages/vendors-single/SendEmail.tsx

MdAsifHossainNadim committed:
enhance: Remove vendorStats from tab components for simplified data handling

MdAsifHossainNadim committed:
enhance: Update OverviewTab and TabSections to include new slots for vendor data integration

MdAsifHossainNadim committed:
enhance: Add TopProductSkeleton for loading state in OverviewTab and update localization strings

MdAsifHossainNadim committed:
enhance: Integrate NoInformation component for empty state in OverviewTab

MdAsifHossainNadim committed:
enhance: Refactor import statement for NoInformation component in OverviewTab

MdAsifHossainNadim committed:
enhance: Update OverviewTab to handle empty product image and category states

Aunshon committed:
Admin dashboard general tab, withdraw tab, verification tab, subscription tab. (#2772)

* enhance: Verification tab & withdraw tab done.

* enhance: General tab done.

* Moded verification codes to pro

* enhance: Remove unused tabs from vendor dashboard

* enhance: Add dokan-react-components dependency and localize frontend script in admin dashboard

* Moved eu fields to pro.

* enhance: Refactor GeneralTab to use SlotFillProvider and PluginArea

* enhance: Add NoInformation component export to index

* enhance: Rename NoInformation component file for improved structure

* Moved subscription tab component to pro

* enhance: Improve dependency handling and add fallback for country/state retrieval

* enhance: Reorder tab positions in TabSections for improved navigation

* enhance: Update GeneralTab and NoInformation components for improved default value handling and localization

MdAsifHossainNadim committed:
Merge pull request #2768 from getdokan/enhance/vendor-overview-redesign-with-tabs-ui

enhance: Vendor overview page re-design with tabs UI.

Aunshon committed:
enhance: Adjust responsive grid classes in GeneralTab for better layout on small screens

Aunshon committed:
enhance: Implement Popover component for vendor redesign

Aunshon committed:
enhance: Implement send email button and popup for vendor redesign

Aunshon committed:
enhance: Refactor payment methods handling and add new payment forms for vendor settings

Aunshon committed:
enhance: Add payment method descriptions and improve layout in WithdrawPaymentRow

Aunshon committed:
enhance: Add nonce to dashboard data and improve NoInformation component styling

Aunshon committed:
enhance: Add customizable class names for NoInformation component elements

Aunshon committed:
enhance: Add vendor creation and editing functionality with form validation

Aunshon committed:
enhance: Add success toast notification and extend form with customizable slots for vendor creation

Aunshon committed:
enhance: Remove unused imports from Create component for cleaner code

Aunshon committed:
enhance: Add LoaderCircle icon to Form component and import Card in Create component

Aunshon committed:
enhance: Update vendor creation page with improved navigation and button functionality

Aunshon committed:
Merge branch 'develop' into enhance/react-add-new-vendor

# Conflicts:
#	src/admin/dashboard/components/Dashboard.tsx

Aunshon committed:
enhance/react-edit-vendor (#2842)

* enhance: Add vendor editing functionality with form validation and navigation

* enhance: Improve layout and add social information slots in vendor edit form

* feat: add CategoryBasedCommissionPure component with hierarchical category management and commission input handling

* feat: rename CombineInputPure to FixedCommissionInput for clarity and consistency

* enhance: Refactor vendor form handling with validation and API request utilities

* feat: refactor components to use lucide-react icons and improve type definitions

* feat: enhance FixedCommissionInput to conditionally render title and description

* feat: optimize className handling in FixedCommissionInput using twMerge

* enhance: Refactor vendor form handling with validation and API request utilities

* feat: enhance vendor form with commission type selection and modal confirmation

* feat: export additional components including DokanModal, DataViews, and error layouts

* feat: integrate commission types via filters in vendor form

---------

Co-authored-by: Osman Goni Sufy <47870515+osmansufy@users.noreply.github.com>

Aunshon committed:
fix: update success message for vendor edit action

shohag121 committed:
WIP: add withdraw management with new admin dashboard page

MdAsifHossainNadim committed:
enhance: Update webpack configuration for Dokan dependency mapping

MdAsifHossainNadim committed:
enhance: Update webpack configuration for Dokan dependency mapping

MdAsifHossainNadim committed:
enhance: Update comments for React utilities and hooks registration in Assets.php

shohag121 committed:
Merge remote-tracking branch 'upstream/enhance/load-dokan-dependency-package-using-webpack' into feat/dokan-widthraw-admin-list

MdAsifHossainNadim committed:
enhance: Refactor import statements and improve code formatting in Bootstrap.js

shohag121 committed:
Merge remote-tracking branch 'upstream/enhance/load-dokan-dependency-package-using-webpack' into feat/dokan-widthraw-admin-list

shohag121 committed:
feat: enhance withdraw admin dashboard with improved formatting, styling, and eligibility conditions

- Improved handling for `dokanAdminDashboard.currency` fallback in `Accounting.ts`.
- Implemented `processDetails` for detailed render of withdrawal methods.
- Replaced custom buttons with `DokanButton` for better consistency.
- Added eligibility conditions for bulk actions (approve, cancel, delete).
- Introduced `formatPrice` utility for consistent price formatting.
- Updated table styling with new borders and rounded corners in SCSS.
- Corrected TSPath alias in `tsconfig.json`.

MdAsifHossainNadim committed:
enhance: Refactor withdrawal actions to utilize modal for approval, cancellation, and deletion

MdAsifHossainNadim committed:
enhance: Improve withdrawal admin list with filter options and tooltip for details

MdAsifHossainNadim committed:
enhance: Add option to download PayPal mass payment file in withdrawal admin list

MdAsifHossainNadim committed:
enhance: Update withdrawal actions with new icons and improve button styling

MdAsifHossainNadim committed:
Merge branch 'develop' into feat/dokan-widthraw-admin-list

Aunshon committed:
feat: add DokanSelect component with customizable styles and indicators

Aunshon committed:
enhance: update DokanSelect component to improve cursor styles for indicators and options

Aunshon committed:
enhance: update indicator separator visibility in DokanSelect component

Aunshon committed:
enhance: improve component composition and style precedence in DokanSelect

shohag121 committed:
Merge branch 'develop' into feat/dokan-widthraw-admin-list

Aunshon committed:
feat: add DokanAsyncSelect and VendorAsyncSelect components for enhanced vendor selection

Aunshon committed:
feat: add ProductAsyncSelect component for enhanced product selection

Aunshon committed:
feat: add WcDateRangePicker component for date range selection

Aunshon committed:
feat: add WpDateTimePicker component for date and time selection

Aunshon committed:
feat: refactor ProductAsyncSelect for improved loading and option handling

Aunshon committed:
feat: add OrderAsyncSelect component for enhanced order selection

Aunshon committed:
feat: add SearchInput component for enhanced search functionality

Aunshon committed:
feat: add CouponAsyncSelect component for coupon selection

Aunshon committed:
feat: add Test component with various Dokan UI elements and functionality

Aunshon committed:
feat: remove DokanSelect export from index.tsx

Aunshon committed:
feat: remove DokanSelect component from Test.tsx

Aunshon committed:
feat: enhance Test component with ProductAsyncSelect and VendorAsyncSelect

Aunshon committed:
feat: remove unused imports from Test component

Aunshon committed:
feat: rename DokanAsyncSelect to AsyncSelect and update imports

Aunshon committed:
feat: rename DokanAsyncSelect to AsyncSelect and update imports

Aunshon committed:
feat: add Select component and update AsyncSelect usage in Test component

Aunshon committed:
feat: refactor AsyncSelect and related components to use BaseSelectProps for improved type safety

Aunshon committed:
feat: enable multi-select functionality in Test component

Aunshon committed:
feat: integrate SimpleInput for date input in WcDateRangePicker, WpDatePicker, and WpDateTimePicker components

Aunshon committed:
feat: add documentation for AsyncSelect, CouponAsyncSelect, OrderAsyncSelect, ProductAsyncSelect, VendorAsyncSelect, WpDateTimePicker, SearchInput, and Select components

Aunshon committed:
Merge branch 'develop' into feat/dokan-select-box

Aunshon committed:
Merge branch 'develop' into update/admin-vendor-profile

# Conflicts:
#	src/admin/dashboard/components/Dashboard.tsx
#	src/admin/dashboard/index.tsx

Aunshon committed:
Merge branch 'update/admin-vendor-profile' into enhance/implement-send-email-button-and-popup-for-vendor-redesign

# Conflicts:
#	src/components/index.tsx

Aunshon committed:
Merge pull request #2766 from getdokan/enhance/implement-send-email-button-and-popup-for-vendor-redesign

enhance: Implement send email button and popup for vendor communication. (Base Vendor Redesign)

Aunshon committed:
Merge pull request #2889 from getdokan/update/admin-vendor-profile

Update/admin vendor profile

Aunshon committed:
Merge branch 'update/dokan-admin-vendor' into enhance/react-add-new-vendor

# Conflicts:
#	src/admin/dashboard/components/Dashboard.tsx
#	src/components/commission/utils.ts
#	src/components/index.tsx

Aunshon committed:
Merge pull request #2819 from getdokan/enhance/react-add-new-vendor

React add new vendor

Aunshon committed:
feat: add prefetch and strict validation options to AsyncSelect components

Aunshon committed:
feat: update AsyncSelect components to improve fetching logic and add unique instance IDs

Aunshon committed:
feat: enhance AsyncSelect components with unique merging logic for prefetched options

Aunshon committed:
feat: implement skip merging logic for AsyncSelect components on dependency changes

Aunshon committed:
feat: replace Test component with Status component in routing

Aunshon committed:
feat: add Vendors management page and related components

Aunshon committed:
Merge branch 'feat/dokan-select-box' into update/dokan-vendor-admin-list

Aunshon committed:
feat: enhance vendor admin list with avatar display and action buttons

Aunshon committed:
feat: update vendor admin list with improved URL handling and status display

Aunshon committed:
feat: add confirmation modal for vendor approval and disable actions

Aunshon committed:
feat: add selection state management for bulk actions in vendor admin list

Aunshon committed:
feat: rename WcDateRangePicker and WpDateTimePicker components to DateRangePicker and DateTimePicker

Aunshon committed:
feat: enhance AsyncSelect components with Tailwind CSS styling and improve option label formatting

Aunshon committed:
feat: update documentation for Props API and enhance WpDatePicker with unique instance IDs

Aunshon committed:
feat: enhance AsyncSelect component with icon support and update documentation

Aunshon committed:
feat: update ProductAsyncSelect endpoint and enhance SearchInput with stable debouncing

Aunshon committed:
feat: remove onClose handler from DateRangePicker component

Aunshon committed:
feat: replace Test component with Status in Dokan routes

Aunshon committed:
Finish select components

Aunshon committed:
feat: enhance AsyncSelect components with onMenuOpen handling and loading optimization

Aunshon committed:
Merge branch 'develop' into feat/dokan-select-box

Aunshon committed:
Merge branch 'feat/dokan-select-box' into update/dokan-vendor-admin-list

Aunshon committed:
feat: add filter button and toggle functionality to Vendors page

Aunshon committed:
feat: add seller badge and subscription filters to Vendors page

Aunshon committed:
feat: improve loading states and optimize rendering for vendor fields

Aunshon committed:
feat: add SellerBadgeSelect component and update vendor list layout

Aunshon committed:
Merge branch 'develop' into update/dokan-admin-vendor

Aunshon committed:
fix: update import paths for components to use relative paths

shohag121 committed:
Merge branch 'develop' into feat/dokan-widthraw-admin-list

# Conflicts:
#	includes/Admin/Dashboard/Dashboard.php

shohag121 committed:
feat: enhance withdraw admin page with new filters and UI updates

- Added `VendorAsyncSelect` and `DateRangePicker` components for improved filtering.
- Updated SCSS to fix z-index and add relative positioning.
- Refactored filter handling logic for better clarity and usability.

Aunshon committed:
feat: enhance vendor filtering with dynamic fields and improved query handling

Aunshon committed:
Merge branch 'develop' into update/dokan-vendor-admin-list

shohag121 committed:
feat: add payment method filter and improve withdraw filtering UI

- Introduced `AsyncSelect` with payment method filter support.
- Enhanced UI components for filtering, including updated placeholders and button styles.
- Added `loadPaymentMethods` function to dynamically fetch payment methods.
- Improved filter reset functionality with additional cleanup actions.

shohag121 committed:
fix: bind payment method value to AsyncSelect in withdraw filter UI

shohag121 committed:
feat: integrate WooCommerce export system with new Dokan export controllers

- Added `ExportController` and `WithdrawExportController` to manage export functionality.
- Registered `WithdrawExportController` with WooCommerce under `withdraws` type.
- Created comprehensive usage documentation for the export controller.
- Enhanced `REST/Manager` to support export controllers seamlessly.

shohag121 committed:
feat: enable withdraw export functionality with enhanced WooCommerce integration

- Added support for withdraw data export via WooCommerce export system.
- Introduced `register_data_endpoint` method to map withdraw export endpoint.
- Updated `ExportController` and `WithdrawExportController` for endpoint structure alignment.
- Enhanced admin withdraw page with bulk export options and improved pagination.

mrabbani commented on src/admin/dashboard/pages/vendors-single/InformationTabs/Withdraw/MethodAction.tsx:
```diff
@@ -0,0 +1,187 @@
+        );
+    }
+
+    const getPaymentData = ( currentMethod ) => {
+        const data = vendor?.payment[ currentMethod ];
+
+        if ( ! data ) {
```
Consider this suggestion for null value

mrabbani commented on src/admin/dashboard/pages/vendors-single/InformationTabs/WithdrawTab.tsx:
```diff
@@ -0,0 +1,130 @@
+        );
+    };
+
+    useEffect( () => {
+        apiFetch( {
+            path: addQueryArgs( 'dokan/v1/settings', {
+                vendor_id: vendor?.id,
+            } ),
+        } )
+            .then( ( data ) => {
+                setIsLoading( false );
+                setSettings( data );
+            } )
+            .catch( () => {
+                setIsLoading( false );
+                setSettings( {} );
+            } );
+    }, [] );
```
Consider the suggestion for dependency.

mrabbani commented on src/admin/dashboard/pages/vendors-single/Skeletons/SkelitonLoader.tsx:
```diff
@@ -0,0 +1,89 @@
+const SidebarSkeleton = () => (
```
Pls check the suggested refactor.

shohag121 replied:
changed accordingly

mrabbani reviewed (changes requested):
Replace the placeholder `ex-` to `e.g. `

mrabbani commented:
Pls resolve the conflict.

Aunshon committed:
Merge branch 'develop' into update/dokan-admin-vendor

# Conflicts:
#	src/components/Popover.tsx
#	src/components/index.tsx

Aunshon committed:
Fix comments

Aunshon committed:
Fix comments

Aunshon committed:
fix: standardize i18n domain to 'dokan-lite' and update component imports

Aunshon committed:
fix: standardize i18n domain to 'dokan-lite' and update component imports

Aunshon committed:
fix: standardize i18n domain to 'dokan-lite' and update component imports

Aunshon committed:
fix: standardize i18n domain to 'dokan-lite' and update component imports

Aunshon committed:
refactor: rename `CategoryBasedCommissionPure` to `CategoryBasedCommission` and update all references

Aunshon committed:
refactor: rename `CategoryBasedCommissionPure` to `CategoryBasedCommission` and update all references

Aunshon committed:
Update Edit.tsx functionality.

shohag121 committed:
chore: apply WordPress hooks for social platforms and payment gateways mapping, and fix SkeletonLoader component rename

shohag121 committed:
Merge branch 'refs/heads/develop' into feat/dokan-widthraw-admin-list

# Conflicts:
#	src/admin/dashboard/style.scss

Aunshon committed:
Merge branch 'develop' into update/dokan-admin-vendor

# Conflicts:
#	includes/Admin/Dashboard/Dashboard.php

shohag121 committed:
Merge remote-tracking branch 'refs/remotes/upstream/develop' into feat/dokan-widthraw-admin-list

# Conflicts:
#	includes/DependencyManagement/Providers/AdminDashboardServiceProvider.php
#	includes/REST/Manager.php

shohag121 committed:
feat: implement reusable admin data table component and enhance withdraw admin UI

- Created `AdminDataViewTable` component for reusable, filterable, and responsive data tables.
- Integrated new data table in withdraw admin page for better UX.
- Added new SCSS styles to support component styling.
- Enhanced withdraw page functionality with improved tabs, export options, and filtering.

shohag121 committed:
feat: add filter button enhancements and improve admin filters behavior

- Introduced `openSelectorSignal` and `buttonRef` states in `AdminDataViewTable` for better filter button management.
- Enhanced `AdminFilter` with new props: `openOnMount`, `openSelectorSignal`, `onFirstFilterAdded`, and `buttonPopOverAnchor` for improved flexibility.
- Adjusted filter popover logic for dynamic anchoring and better user experience.
- Updated withdraw admin page tab icons for better readability.

shohag121 committed:
feat: auto-hide filters panel when no active filters

- Added `hasActiveFilters` state in `AdminDataViewTable` to track active filters.
- Implemented logic to auto-hide filter area when no filters are active.
- Enhanced `AdminFilter` with `onActiveFiltersChange` prop for notifying parent components about active filter count changes.
- Improved filter button behavior by conditionally toggling filter visibility.

shohag121 committed:
feat: improve withdraw filters UX with enhanced clearing and removal functionality

- Refactored `clearSingleFilter` method to improve filter clearing logic for vendors, payment methods, and date ranges.
- Added UI for removing individual active filters in `AdminFilter` using a close button with `lucide-react` `X` icon.
- Updated filter management to ensure state consistency and reset view pagination on filter changes.

shohag121 committed:
feat: enhance AdminDataViewTable with new empty state and styles

- Introduced reusable `ListEmpty` component to standardize empty state UI.
- Added support for customizable empty state (`empty`, `emptyIcon`, `emptyTitle`, `emptyDescription`) in `AdminDataViewTable`.
- Updated `dataviews` SCSS with improved layouts, spacing, and styles for empty states and table paddings.
- Enhanced withdraw admin page with updated tab and status badge styles for better readability.
- Added comprehensive documentation for `AdminDataViewTable` usage in `frontend/admin-dataview-table.md`.

shohag121 committed:
feat: add bulk action toolbar and improve data view styles

- Added `BulkActionToolbar` component in `AdminDataViewTable` for managing bulk actions.
- Updated SCSS to include box-shadow for `.dataviews-wrapper` to enhance visual hierarchy.

shohag121 committed:
refactor: update REST Manager to simplify controller loading

- Commented out redundant code checking/loading `WithdrawExportController`.
- Added `ExportController` to `get_registered_controllers` method for enhanced export functionality.

shohag121 committed:
fix: update bulk action callback to map item IDs

- Modified `handleBulkAction` usage to map `items` to their `id` values for better data handling in bulk PayPal actions.

shohag121 committed:
refactor: simplify withdraw export argument structure

- Updated `report_args` to directly use `ids` instead of `include` for better clarity and consistency.

shohag121 committed:
fix: add per_page export argument and ensure integer pagination headers

- Added `per_page` argument to withdraw export API for controlling items per page.
- Ensured `X-WP-Total` and `X-WP-TotalPages` headers return integers in withdraw export responses.

shohag121 committed:
fix: update filter panel visibility logic in AdminDataViewTable

- Applied `!hidden` class for proper handling of `showFilters` state to ensure consistent filter panel visibility.

shohag121 committed:
fix: improve withdraw modal UI and localization support

- Added `_n` import for handling singular/plural translations in withdraw modal.
- Refined modal styling with better section separation using borders.
- Updated `dialogTitle` to display amount and store name for clarity.
- Replaced repetitive header code with streamlined component structure.

Aunshon committed:
Merge branch 'develop' into update/dokan-vendor-admin-list

# Conflicts:
#	includes/DependencyManagement/Providers/AdminDashboardServiceProvider.php
#	src/admin/dashboard/components/Dashboard.tsx

Aunshon committed:
update/link-dokan-admin-vendor (#2926)

* refactor: replace `router-link` with `<a>` tags in Vendors.vue and update navigation logic

* feat: add support for admin commission settings when creating a vendor.

* refactor: replace `MaskedInput` with `DebouncedInput` in CommissionInputs for improved performance

* refactor: centralize DebouncedInput component and update references accordingly

* refactor: extract StoreName to a separate component and add filter for customization in HeaderCard

* feat: add PluginArea to InfoCard for extending vendor details via plugins

* refactor: restructure vendor form sections into distinct components for better maintainability

* feat: add discard changes functionality to Edit vendor page and update button labels

* feat: add unsaved changes warning when navigating away from Edit vendor page

* feat: add confirmation modal for unsaved changes on Edit vendor page navigation

* fix: ensure tab component is rendered correctly in Form component

* refactor: improve layout and styling of vendor forms and info sections

* refactor: update tab navigation implementation and improve component rendering

* refactor: change categories state initialization to an object and update data handling

* fix: adjust vendor management for legacy page handling and update minor UI classes

---------

Co-authored-by: Md Asif Hossain Nadim <devianadim@gmail.com>

Aunshon committed:
Merge branch 'feat/dokan-widthraw-admin-list' into update/dokan-vendor-admin-list

# Conflicts:
#	includes/DependencyManagement/Providers/AdminDashboardServiceProvider.php
#	src/admin/dashboard/components/Dashboard.tsx

shohag121 committed:
fix: enhance withdraw export status handling and data updates

- Updated `checkStatus` function to use Promises for improved async handling and status check resolution.
- Added `payable` field to withdraw export data for more comprehensive reporting.
- Refined `AdminDataViewTable` tab selection logic to reset selection state.
- Updated dependencies to the latest versions for improved stability and features.

shohag121 committed:
fix: filter PayPal items in bulk action callback

- Updated `handleBulkAction` to process only items with the `paypal` method before mapping their IDs.

Aunshon committed:
Merge branch 'feat/dokan-widthraw-admin-list' into update/dokan-vendor-admin-list

Aunshon committed:
refactor: enhance vendor admin list with dynamic filters and improved UI elements

- Updated filters to allow dynamic field injection and better customization.
- Improved UI for status tabs and filters with consistent theming.
- Refactored filter logic and replaced redundant methods for better clarity.
- Integrated search and tab selection into the main table component for enhanced usability.

Aunshon committed:
refactor: remove unused components and redundant state logic in Vendors page

- Removed `DokanTab` and `Filter` components.
- Deleted `showFilters` state and unnecessary `useEffect` for filter visibility.
- Simplified and cleaned up markup and filter methods for improved readability.

Aunshon committed:
Merge branch 'develop' into update/dokan-vendor-admin-list

MdAsifHossainNadim committed:
Merge branch 'develop' into feat/dokan-widthraw-admin-list

shohag121 committed:
fix: enhance withdraw modal logic and UI

- Added `selectedAction` state to manage action selections in the withdraw modal.
- Improved action initialization for the "view" modal based on item status.
- Simplified conditional rendering and cleaned up modal structure for better readability.
- Refined button handling, ensuring appropriate action execution and modal behavior.

Aunshon committed:
Merge branch 'develop' into feat/dokan-widthraw-admin-list

Aunshon committed:
Merge branch 'feat/dokan-widthraw-admin-list' into update/dokan-vendor-admin-list

Aunshon committed:
feat: add 'vendors' to supported keys in PanelSwitch filter

Aunshon committed:
Merge branch 'develop' into update/dokan-admin-vendor

Aunshon committed:
feat: add 'vendors' to supported keys in PanelSwitch filter

MdAsifHossainNadim committed:
refactor: Update hash handling in PanelSwitch to enhance URL change detection and reactivity

shohag121 committed:
fix: add 'withdraw' to panel switch supported keys filter

- Extended `supportedKeys` filter to include 'withdraw' for better panel switch handling.

MdAsifHossainNadim committed:
Merge branch 'feat/dokan-widthraw-admin-list' into update/dokan-vendor-admin-list

# Conflicts:
#	src/admin/panel-switcher/PanelSwitch.tsx

MdAsifHossainNadim committed:
refactor: Remove unused `shouldRender` state and legacy redirect logic from PanelSwitch and LegacySwitcher

Aunshon committed:
fix: update vendor navigation path in Vendors page

Aunshon committed:
refactor: replace `wp.hooks.applyFilters` with `applyFilters` and update `wp_localize_script` usage

- Used `@wordpress/hooks`'s `applyFilters` for consistency.
- Replaced `wp_localize_script` with `wp_add_inline_script` for better script handling.

Aunshon committed:
Merge branch 'update/dokan-vendor-admin-list' into update/dokan-admin-vendor

# Conflicts:
#	includes/Admin/Dashboard/Dashboard.php
#	src/admin/dashboard/components/Dashboard.tsx
#	src/admin/panel-switcher/PanelSwitch.tsx
#	src/components/index.tsx
#	tsconfig.json

Aunshon committed:
fix: set `dokanAdminDashboard` as a global window object for script access

Aunshon committed:
fix: update vendor creation navigation path in Vendors page

Aunshon committed:
Merge branch 'develop' into update/dokan-admin-vendor

# Conflicts:
#	includes/Admin/Dashboard/Dashboard.php
#	src/admin/dashboard/components/Dashboard.tsx
#	src/admin/dashboard/pages/vendors.tsx
#	src/admin/dashboard/pages/withdraw/index.tsx
#	src/components/index.tsx

Aunshon committed:
refactor: replace `AsyncSearchableSelect` with `Select` in Vendor form

- Switched to using the `Select` component for country and state fields.
- Removed `defaultOptions` in favor of `options` for better compatibility.

Aunshon committed:
feat: add delay prop to store name and email fields in Vendor form

Aunshon committed:
fix: update vendor navigation links to use the correct dashboard path

Aunshon committed:
fix: downgrade `@wordpress/dataviews` to version 6.0.0 for compatibility
